### PR TITLE
[Refactor][EAGLE] 6/N route mtp to eagle except pcp/dcp+mtp

### PR DESCRIPTION
**Pull request by lilinsiman**
### What this PR does / why we need it?

Overview: This pull request refactors speculative decoding for Eagle and MTP proposers on Ascend hardware. It fixes a bug related to draft_attn_metadatas being lost, migrates the lmhead feature, and adds routing logic in MtpProposer.

Details:
1. Migrated the lmhead feature from mtp to eagle and normalized it in eagle_proposer.
2. Fixed the bug where draft_attn_metadatas was lost after enabling eagle mode in the merge graph.
3. Added the routing for pcp and disable padded drafter batch; in mtp mode, if pcp and disable padded drafter batch are not enabled, the normalized file eagle_proposer will be used.

RFC: https://github.com/vllm-project/vllm-ascend/issues/5467

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
ut and test

- vLLM version: v0.14.1
- vLLM main: https://github.com/vllm-project/vllm/commit/dc917cceb877dfd13f98c538c4c96158047d98bd

### CODE REVIEW

File: vllm_ascend/spec_decode/eagle_proposer.py [hunk 1]
```diff
@@ -701,10 +718,25 @@ def _run_merged_draft(self,
...
+                    last_token_indices,
+                    (0, max_num_reqs_across_dp - num_indices),
+                )
+
+            sample_hidden_states = last_hidden_states[last_token_indices]
+            logits = self.model.compute_logits(sample_hidden_states)
+
+            if lmhead_tp_enable() and num_indices < logits.shape[0] and not is_dummy:
+                logits = logits[:num_indices]
+                last_token_indices = last_token_indices[:num_indices]
```
[the remark this answers is not in the document]
Comment (lilinsiman) [reply vllm_ascend/spec_decode/eagle_proposer.py:748]: This function still has some issues that need to be fixed, and there are many such issues in the file. In the next phase, the code of the entire file will be uniformly revised, and duplicate code will be abstracted into functions.
